app/routes/_layout+/assets.$assetId.tsx

```diff
@@ -143,7 +147,36 @@ export async function action({ context, request, params }: ActionFunctionArgs) {
           senderId: authSession.userId,
         });
 
-        return redirect(`/assets`);
+        return redirect("/assets");
+      }
+
+      case "relink-qr-code": {
+        const { newQrId } = parseData(
+          formData,
+          z.object({ newQrId: z.string() })
+        );
+
+        await Promise.all([
+          claimQrCode({
```

Comment:
I think you might have to create a new function for this called `relinkAssetCode` that calls `updateAssetQrCode` but also does some extra checks and validations and has its own error handling because the one from claimQrCode doesnt give the correct errors.